### PR TITLE
Support appendAssemblySize() in AbstractAssembly

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
I've tried porting lll over to use AbstractAssembly and this was needed. We haven't designed yet how to handle "Julia objects" (sub assemblies, etc.) in AbstractAssembly, but this is one step towards that.

File: libjulia/backends/evm/EVMAssembly.cpp [hunk 1]
```diff
@@ -145,6 +147,12 @@ void EVMAssembly::appendReturnsub(int _returns, int _stackDiffAfter)
 
 eth::LinkerObject EVMAssembly::finalize()
 {
+	size_t bytecodeSize = m_bytecode.size();
+	solAssert(uint64_t(bytecodeSize) < (uint64_t(1) << (8 * assemblySizeReferenceSize)), "Bytecode too big.");
+	for (auto const& ref: m_assemblySizePositions)
+		for (size_t i = 0; i < assemblySizeReferenceSize; i++)
```
Comment (chriseth) [inline libjulia/backends/evm/EVMAssembly.cpp]: Can you write a helper for inserting big endian numbers? (same code is also used below)

**Review by chriseth — CHANGES_REQUESTED** [review]

File: libjulia/backends/evm/EVMAssembly.cpp (hunk 1 above)
Comment (axic) [reply libjulia/backends/evm/EVMAssembly.cpp]: Arbitrary size?

File: libjulia/backends/evm/EVMAssembly.cpp [hunk 2]
```diff
@@ -32,6 +32,8 @@ namespace
 {
 /// Size of labels in bytes. Four-byte labels are required by some EVM1.5 instructions.
 size_t constexpr labelReferenceSize = 4;
+
+size_t constexpr assemblySizeReferenceSize = 4;
```
Comment (axic) [inline libjulia/backends/evm/EVMAssembly.cpp:36]: @chriseth any better naming suggestion here?

File: libjulia/backends/evm/EVMAssembly.cpp (hunk 1 above)
Comment (chriseth) [reply libjulia/backends/evm/EVMAssembly.cpp]: If also fine with forcing everything to four bytes for now.

File: libjulia/backends/evm/EVMAssembly.cpp (hunk 2 above)
Comment (chriseth) [reply libjulia/backends/evm/EVMAssembly.cpp:36]: Nothing wrong about the name :-)

File: libjulia/backends/evm/EVMAssembly.cpp (hunk 1 above)
Comment (axic) [reply libjulia/backends/evm/EVMAssembly.cpp]: Added one with u256, but we could change it to size_t or even a template.

**Review by chriseth — APPROVED** [review]

**Pull request merged**